pkg/controller/vitessshardreplication/repair_replication.go

```diff
@@ -151,7 +151,13 @@ func (r *ReconcileVitessShard) canRepairReplication(ctx context.Context, vts *pl
 		return false, fmt.Errorf("failed to execute query against master tablet %v: %v", vts.Status.MasterAlias, err)
 	}
 	if masterReadOnly {
-		return true, nil
+		vts.Status.SetConditionStatus(planetscalev2.ReadOnlyMasterConditionType, corev1.ConditionTrue, "MasterReadOnlyDetected", "Master ReadOnly State Detected.")
+		shouldFix := masterRdOnlyShouldBeFixed(vts)
+		if shouldFix {
+			return true, nil
+		}
+	} else {
+		vts.Status.SetConditionStatus(planetscalev2.ReadOnlyMasterConditionType, corev1.ConditionFalse, "MasterNotReadOnly", "Master is not in ReadOnly State.")
```

Comment:
Maybe I'm misunderstanding, but why do we have a false condition for the master not being in read only? Shouldn't we just need to check that the master doesn't have a "MasterReadOnly" condition?

Reply:
We could either clear the condition entirely from the map, or set it to ConditionFalse. The convention as far as I know is to set the conditions to false when they are false rather than clear them entirely.